textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexInfo.java

```diff
@@ -78,5 +78,98 @@ static RegexInfo anyChar() {
 		regexInfo.emptyable = false;
 		return regexInfo;
 	}
+	
+	/**
+	 * This function simplifies the regexInfo. <br>
+	 * If there the "exact" set gets too large, it will add exact to the query tree,
+	 * and move relevant pieces into prefix and suffix. <br>
+	 * 
+	 * Simplification is performed under three circumstances. <br>
+	 * 1, the size ofexact set is larger than a threshold. <br>
+	 * 2, the minimum length of strings in exact is larger than {@code gramLength+1}, 
+	 *    if we do not set {@code force = true}. <br>
+	 * 3, the minimum length of strings in exact is larger than {@code gramLength}, 
+	 *    if we set {@code force = true}. <br>
+	 * @param force
+	 */
+	RegexInfo simplify(boolean force) {
+		TranslatorUtils.removeDuplicateAffix(exact, false);
+		
+		if ( exact.size() > TranslatorUtils.MAX_EXACT_SIZE ||
+			( TranslatorUtils.minLenOfString(exact) >= GramBooleanQuery.gramLength && force) ||
+			TranslatorUtils.minLenOfString(exact) >= GramBooleanQuery.gramLength + 1){
+			// Add exact to match (query tree)
+			// Transfer information from exact to prefix and suffix
+			match.add(exact);
+			for (String str: exact) {
+				if (str.length() < 3) {
+					prefix.add(str);
+					suffix.add(str);
+				} else {
+					prefix.add(str.substring(0, 3));
+					suffix.add(str.substring(str.length()-3, str.length()));
+				}
+			}
+			exact.clear();
+		}
+		
+		// Add information in prefix and suffix to match
+		if (exact.isEmpty()) {
+			simplifyAffix(prefix, false);
+			simplifyAffix(suffix, true);
+		}
+		
+		return this;
+	}
+	
+	/**
+	 * simplifyAffix reduces the size of the given set (either prefix or suffix).
+	 * If the set gets too big, it moves information in prefix/suffix into match query.
+	 * @param strList
+	 * @param isSuffix indicates given string list is suffix list or not
+	 */
+	void simplifyAffix(List<String> strList, boolean isSuffix) {
+		TranslatorUtils.removeDuplicateAffix(strList, isSuffix);
+		
+		// Add the current prefix/suffix set to match query.
+		match.add(strList);
+		
+		// This loop cuts the length of prefix/suffix. It cuts all
+		// strings longer than 3, and continues to cut strings
```

Comment:
The comment has not been customised yet... It should be replaced by `GramBooleanQuery.gramLength`.